scripts/test_matrix.py

```diff
@@ -71,220 +42,126 @@ def cleaner(env_path):
-    """Check a given environment's version of bokeh.
-    """
-    if os.name == 'nt':
-        command_string = 'activate %s & python -c "import sys, bokeh; sys.exit(0 if bokeh.__version__ == %r else 1)"' % (env_name, expected_ver)
-    else:
-        command_string = 'source activate %s; python -c "import sys, bokeh; sys.exit(0 if bokeh.__version__ == %r else 1)"' % (env_name, expected_ver)
-
-    result = call_wrapper(command_string, shell=True)
-
-    return result == 0
-
-
-def run_tests(env_name):
-    """Run py.test in a given environment.  Writes results to a tmpfile that will be returned as a string
-    in the event of a failure.
-    """
-
-    test_failure = ''
-    file_name  = "tmpfile.txt"
-    tmpfile = open(file_name, "w+")
-
-    if os.name == 'nt':
-        command_string = 'activate %s & py.test -s --cov=bokeh"' % env_name
-    else:
-        command_string = 'source activate %s; py.test -s --cov=bokeh"' % env_name
-
-    result = call_wrapper(command_string, shell=True, stderr=tmpfile)
-
-    tmpfile.close()
-
-    if result != 0:
-        with open(file_name, "r") as tmpfile:
-            test_failure = (tmpfile.read())
-
-    os.remove(file_name)
-
-    return result == 0, test_failure
-
-
-def server_check(env_name):
-    pass
-    # Open subprocess, run bokeh server
-    # Check return code, if 0, True, if 1, False
-    # Kill subprocess
-
-
-def logger(failure_list, logfile='logfile.txt'):
-    """Log items in a list of errors to a logfile.
-    """
-
-    while os.path.exists(logfile):
-        logfile = logfile.split('-')
-        if len(logfile) == 1:
-            logfile.append('2')
-        else:
-            logfile[1] = str(int(logfile[1]) + 1)
-        logfile = '-'.join(logfile)
-
-    with open(logfile, 'w') as log:
-        for failure in failure_list:
-            log.write(failure)
-
-
 if __name__ == '__main__':
 
     parser = get_parser()
     ops = parser.parse_args()
-    OPTIONS["dry-run"] = ops.dry_run
 
-    preversion = ops.previous
-    current_version = ops.version
+    ver = ops.version
 
     envs = {
         "py27_conda_clean"    : {
-            "init"    : "python=2.7 pytest pytest-cov mock",
+            "init"    : "python=2.7 pytest mock",
             "install" : '; '.join([
                     # install latest version from dev channel
                     "conda install --yes -c bokeh/channel/dev bokeh",
                     # install dependencies needed for testing
+                    "conda install --yes  -c jrderuiter pytest-cov",
                     "conda install --yes  -c auto websocket-client",
-                    "conda install --yes  -c bokeh pytest pytest-cov mock blaze abstract-rendering beautiful-soup "
-                    "ipython scipy websocket multiuserblazeserver pillow",
+                    "conda install --yes  -c bokeh nose mock blaze abstract-rendering beautiful-soup ipython scipy websocket multiuserblazeserver pillow",
```

Comment:
@Maggie-M do you have the commit ready? If not... I will merge is as an intermediate step, because it will be useful for the upcoming release, and you can make another PR later...